crates/solver/src/liquidity/order_converter.rs

```diff
@@ -88,6 +90,7 @@ impl OrderConverter {
 struct OrderSettlementHandler {
     order: Order,
     native_token: WETH9,
+    allow_unwrap_native_token: bool,
```

Comment:
nit: seems more to the point
```suggestion
    manage_native_token_unwraps: bool,
```